src/Microsoft.TestPlatform.Utilities/CodeCoverageDataAttachmentsHandler.cs

```diff
@@ -58,70 +56,47 @@ public Task<ICollection<AttachmentSet>> ProcessAttachmentSetsAsync(ICollection<A
 
         private string MergeCodeCoverageFiles(IList<string> files, IProgress<int> progressReporter, CancellationToken cancellationToken)
         {
-            if(files.Count == 1)
+            if (files.Count == 1)
             {
                 return files[0];
             }
 
-            string tempFileName = Path.Combine(Path.GetTempPath(), Guid.NewGuid() + CoverageFileExtension);
-            string outputfileName = files[0];
-
-            File.Create(tempFileName).Dispose();
-            var assemblyPath = Path.Combine(Path.GetDirectoryName(typeof(CodeCoverageDataAttachmentsHandler).GetTypeInfo().Assembly.GetAssemblyLocation()), CodeCoverageAnalysisAssemblyName + ".dll");
-
             try
             {
-                cancellationToken.ThrowIfCancellationRequested();
-                Assembly assembly = new PlatformAssemblyLoadContext().LoadAssemblyFromPath(assemblyPath);
-                var type = assembly.GetType(CodeCoverageAnalysisAssemblyName + "." + CoverageInfoTypeName);
-
-                var methodInfo = type?.GetMethod(MergeMethodName);
-
-                if (methodInfo != null)
-                {
-                    IList<string> filesToDelete = new List<string>(files.Count) { tempFileName };
-
-                    for (int i = 1; i < files.Count; i++)
-                    {
-                        cancellationToken.ThrowIfCancellationRequested();
-                        progressReporter?.Report(100 * i / files.Count);
-
-                        cancellationToken.ThrowIfCancellationRequested();
-                        methodInfo.Invoke(null, new object[] { files[i], outputfileName, tempFileName, true });
-
-                        cancellationToken.ThrowIfCancellationRequested();
-                        File.Copy(tempFileName, outputfileName, true);
-
-                        filesToDelete.Add(files[i]);
-                    }
-
-                    cancellationToken.ThrowIfCancellationRequested();
-                    foreach (string fileName in filesToDelete)
-                    {
-                        File.Delete(fileName);
-                    }
-                }
-
+                // Warning: Don't remove this method call.
+                //
+                // We took a dependency on Coverage.CoreLib.Net. In the unlikely case it cannot be
+                // resolved, this method call will throw an exception that will be caught and
+                // absorbed here.
+                this.MergeCodeCoverageFiles(files, cancellationToken);
```

Comment:
It would be great to add IProgress to interface. @fhnaseer is this something that we can add? Does it make sense in our logic?
We can do it as new commits in both repos

Reply:
We can do this. Maybe in next version,